python/xir/parser.py

```diff
@@ -131,7 +131,7 @@ def wires(self, w):
     def var(self, v):
         """Expressions that are strings are considered to be variables. Can be
         substituted by values at a later stage."""
-        self._program._variables.add(v[0])
+        self._program.add_variable(v[0])
```

Comment:
Also below, `self._program._statements.append(args[0])` can be changed to `self._program.add_statements(args[0])`.

Reply:
Right!